src/api.cc

```diff
@@ -362,4 +362,15 @@ TufTarget AkliteClient::GetPendingTarget() const {
   return !pending ? TufTarget() : Target::toTufTarget(*pending);
 }
 
-bool AkliteClient::CompleteInstallation() { return client_->finalizeInstall(); }
+InstallResult AkliteClient::CompleteInstallation() {
+  auto install_completed{client_->finalizeInstall()};
+  InstallResult complete_install_res{InstallResult::Status::Failed, ""};
+  if (install_completed) {
+    if (!client_->isBootFwUpdateInProgress()) {
```

Comment:
I cannot comment on commit message.

Need to add `- Handle boot firmware update` to it.

Reply:
I just accidentally merged two commit during rebase. Anyway, this is draft PR atm, and there are still a lot of to do, so I will restructure commits anyway.